edx_rbac/utils.py

```diff
@@ -54,16 +55,22 @@ def user_has_access_via_database(user, role_name, role_assignment_class, context
     """
     Check if there is a role assignment for a given user and role.
 
-    The role object itself is found via the role_name
+    The role object itself is found via the role_name. The role_assignment_class's get_context() method can return a
+    single context string which could be an ALL_ACCESS_CONTEXT or, incase of multiple user contexts, a list of strings.
+    The context argument is evaluated against the context(s) received from the role_assignment_class while accounting
+    for the ALL_ACCESS_CONTEXT to grant access.
     """
     try:
         role_assignment = role_assignment_class.objects.get(user=user, role__name=role_name)
     except role_assignment_class.DoesNotExist:
         return False
 
     if context:
-        return role_assignment.get_context() in (context, ALL_ACCESS_CONTEXT)
-
+        context_in_database = role_assignment.get_context()
+        if isinstance(context_in_database, string_types):
+            return context_in_database in (context, ALL_ACCESS_CONTEXT)
+        else:  # Multiple context case
+            return context in context_in_database
```

Comment:
so i think this will fail if one the context values in the database is `*`. maybe this should be ```return context in context_in_database or ALL_ACCESS_CONTEXT in context_in_database```